### PR TITLE
Add Game Design Document

### PR DESCRIPTION
**Pull request by marwin89**
### What does this PR?

- This  pr adds the first official game design document for vocaluxe.
- It contains Concept, Technical Specs, Game Design, Style Guide and more.
- It serves as foundation for game programmers and artists.

### Why

- We didn't have a central game design document yet. It's standard in games industry.
- Many game developers ask for this and this helps recruiting and motivating programmers
- Definitive and precise vision for Vocaluxe
- Helps us to stay focused

### More Info

- Helps to complete: #720

### CODE REVIEW

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
+<img alt="Vocaluxe Logo" src="https://github.com/Vocaluxe/Vocaluxe/blob/develop/Output/Themes/Vocaluxe%202024/LogoMain.png" width=500/>
+
+## Concept and Game Design Document
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Should be on the first level, right, so just one '#'.

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+* Sing alone and collect scores
+* Sing with friends together or against each other
+* browse and listen to songs
+* watch music videos
+
+### Player:
+
+*The number of players that can play the game at once*
+
+* 1-4 Players
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: 1-6 players

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+SingStar Celebration (2017), Let’s Sing 2023, UltraStar Deluxe (2008)
+
+### Price:
+
+free (non-commercial & open source), yet music have to be buyed in trusted stores
+
+### Planned Release Date:
+
+1st February 2026 (if possible)
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Let's skip that section. It's just a date, which isn't reachable without developers (or might be earlier with developers=

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+*Camera view from which the player will experience the game:*
+
+* 3rd Person
+* Front, only Screens
+* Fixed Camera, no rotation
+
+### Platform:
+
+Windows, Mac, Linux
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Only windows, mac and linux might be possible but need some more love and attention.

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+
+## TECHNICAL SPECS
+
+### Technical Form:
+
+*Basically there are 2D graphics (flat) and 3D graphics (form):*
+
+* 2D graphics (flat)
+
+### View:
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md:66]: Not relevant?

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+
+Vocaluxe has these following features:
+
+| **Song Collection** | Custom Songs. Limit is 10.000 songs. |
+| :---- | :---- |
+| **License** | Open Source, MIT-License |
+| **Highscore** | For each song, for each difficulty, for each player |
+| **Statistics** | most played songs, songs never played |
+| **In Built Store** | No |
+| **Files Stored** | Locally, optional: Youtube Stream Embed |
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Youtube isn't possible atm.

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+| **Translations (UI)**  | Czech, Dutch, English, French, German, Hungarian, Italian, Portuguese, Spanish, Swedish, Turkish, Chinese, Japanese |
+| **Difficulty levels** | easy, medium, difficult |
+| **Theme support** | yes, each game provides customization of look and feel. description below. |
+| **Playlist support** | yes (.m3u) |
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: It's a custom vocaluxe format if i remember correctly.

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+| **Party modes** | yes: each game has at least 3 different party modes. description below. |
+| **Sing modes** | yes: Versus (Duell), Co-Op (Duet), Co-Op 2 (Pass the mic), Medley Duell |
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: No Co-Op2 and Medley Duell at them moment.

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+| **Freestyle notes support (F)** | yes |
+| **Golden notes support (\*)** | yes |
+| **Jukebox mode** | yes, play Songs without scoring, lyrics, and player profiles |
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Not implemented yet

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+| **Player Profiles** | yes, Player has at least own avatar image, name and individual high score |
+| **Song Menu Style** | Grid View aka "Tile View". |
+| **Song Menu Search and Filter** | yes, filter by: year, genre, artist, favorites |
+| **Gamepad supported** | yes, xbox controller and similiar controller are supported |
+| **Smartphone supported** | yes, webbroswer app, AndroidApp, iOS App |
+| **Automatic Updates** | yes |
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Is that implemented in the installer? don't think so...

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+| **Party Modes** | Yes: Tic Tac Toe Mode Challenge mode |
+| **Song Grid** | 3x4 (=12 songs per page) |
+| **Theming** | choose from a predefined set of 6 themes |
+| **Animation of Notes** | static |
+| **Option for classic karaoke evening** | No \- only supports classic singstar experience |
+| **Max. number of simultaneous singers** | 1-4 Players |
+| **Look & Feel** | like singstar celebration (2017) |
+| **Rap mode** | Yes, Rap-O-Meter: ignore pitch of notes score only rhythm don’t display normal notes show a Rap-O-Meter  |
+| **Practice mode** | No |
+| **Collectables and score** | **collect fame, collect medals gold silver bronze**  |
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Not yet implemented

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+
+* A player is a singer.
+
+* By using his or her voice he can collect points. The better the player gets (and/or the longer the player plays) the more bonus items and features she unlocks during the game.
+
+* A player designs a profile for himself by choosing an Artist name and avatar plus selecting favorite songs and music genres.
+
+### Player Properties
+
+*Each property should mention feedback as a result of the property changing.*
```
Comment (flokuep) [inline Vocaluxe/Doc/GameDesignDocumentVocaluxe.md:270]: Not yet implemented.

**Review by flokuep — CHANGES_REQUESTED** [review]
Thanks for wrapping up! I think we should better distinguish between the current state and a possible version 1.0 state.

**Comment by marwin89** [comment]
> Thanks for wrapping up! I think we should better distinguish between the current state and a possible version 1.0 state.

yes, right now the document describes only the goals and the final state of the game. I will add a another colum for status of current nightly build.

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 1 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: 👍🏻

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 2 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 3 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: I just wrote unknown, depends on developer, ideally in 2025-2027
✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 4 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 5 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md:66]: ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 6 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 7 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: changed playlist to vocaluxe-.xml ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 8 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Made clear in comparison ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 9 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Made clear in comparison ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 11 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md]: Made clear in comparison ✅

File: Vocaluxe/Doc/GameDesignDocumentVocaluxe.md (hunk 12 above)
Comment (marwin89) [reply Vocaluxe/Doc/GameDesignDocumentVocaluxe.md:270]: Made clear in comparison ✅

**Comment by marwin89** [comment]
@flokuep I adressed all your comments

**Review by flokuep — APPROVED** [review]

**Pull request merged**